app/src/components/ChatBot/chat-bot.tsx

```diff
@@ -58,20 +66,251 @@ export default function ChatBot({
   t: TFunction;
   inventoryId: string;
 }) {
-  const {
-    messages,
-    input,
-    handleInputChange,
-    handleSubmit,
-    isLoading,
-    append,
-    reload,
-  } = useChat({
-    api: `/api/v0/chat/${inventoryId}`,
-    initialMessages: [
-      { id: "-1", content: t("initial-message"), role: "assistant" },
-    ],
-  });
+  const [threadId, setThreadId] = useState("");
+  const [userInput, setUserInput] = useState("");
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [createThreadId] = useCreateThreadIdMutation();
+  const [getAllDataSources] = api.useLazyGetAllDataSourcesQuery();
+  const [getUserInventories] = api.useLazyGetUserInventoriesQuery();
+  const [getInventory] = api.useLazyGetInventoryQuery();
+  const toast = useToast();
+
+  const handleError = (error: any, errorMessage: string) => {
+    console.error(errorMessage, error);
```

Comment:
I think we use the  { logger } from "@/services/logger";

Reply:
handleError is for user UI to have a popup to notify the user that the error occured. It is not meant to log any errors in our logging system. That being sad, it might be useful to add logging via "@/services/logger" in general, should be a new ticket then

Reply:
Logger is mainly used in the backend as it has an actual function there (structured logging)
Fine like this